src/main/java/net/syscon/elite/service/MovementUpdateService.java

```diff
@@ -23,55 +25,77 @@
 
     public MovementUpdateService(
             final ReferenceDomainService referenceDomainService,
             final BedAssignmentHistoryService bedAssignmentHistoryService,
             final BookingService bookingService,
             final OffenderBookingRepository offenderBookingRepository,
+            final AgencyInternalLocationRepository agencyInternalLocationRepository,
             final Clock clock) {
         this.referenceDomainService = referenceDomainService;
         this.bedAssignmentHistoryService = bedAssignmentHistoryService;
         this.bookingService = bookingService;
         this.offenderBookingRepository = offenderBookingRepository;
+        this.agencyInternalLocationRepository = agencyInternalLocationRepository;
         this.clock = clock;
     }
 
     @Transactional
     @VerifyBookingAccess
     @HasWriteScope
-    public OffenderBooking moveToCell(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime dateTime) {
+    public OffenderBooking moveToCell(final Long bookingId, final String internalLocationDescription, final String reasonCode, final LocalDateTime dateTime) {
         validateMoveToCell(reasonCode, dateTime);
         final var movementDateTime = dateTime != null ? dateTime : LocalDateTime.now(clock);
-        referenceDomainService.getReferenceCodeByDomainAndCode(CELL_MOVE_REASON.getDomain(), reasonCode, false);
         final var offenderBooking = getActiveOffenderBooking(bookingId);
+        final var internalLocation = getActiveInternalLocation(internalLocationDescription);
 
-        if (offenderBooking.getAssignedLivingUnitId().equals(livingUnitId)) {
+        if (offenderBooking.getAssignedLivingUnitId().equals(internalLocation.getLocationId())) {
             return offenderBooking;
         }
 
-        bookingService.updateLivingUnit(bookingId, livingUnitId);
-        bedAssignmentHistoryService.add(bookingId, livingUnitId, reasonCode, movementDateTime);
+        bookingService.updateLivingUnit(bookingId, internalLocationDescription);
+        bedAssignmentHistoryService.add(bookingId, internalLocation.getLocationId(), reasonCode, movementDateTime);
         return getActiveOffenderBooking(bookingId);
     }
 
     private void validateMoveToCell(final String reasonCode, final LocalDateTime dateTime) {
+        checkReasonCode(reasonCode);
         checkArgument(!StringUtils.isNullOrEmpty(reasonCode), "Reason code is mandatory");
         checkArgument(
                 dateTime == null || dateTime.isBefore(LocalDateTime.now(clock)) || dateTime.isEqual(LocalDateTime.now(clock)),
                 "The date cannot be in the future"
         );
     }
 
+    private void checkReasonCode(String reasonCode) {
+        try {
+            referenceDomainService.getReferenceCodeByDomainAndCode(CELL_MOVE_REASON.getDomain(), reasonCode, false);
+        } catch(EntityNotFoundException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
```

Comment:
meh.. `throw new IllegalArgumentException(e);` would be equivalent

Reply:
That's what I thought... but it turns out the `public Throwable(Throwable cause)` does  `this.detailMessage = cause == null ? null : cause.toString();` and so you get more than just the message.